### PR TITLE
Bump valgrind to 3.12.0

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

**Review by xen0l — APPROVED** [review]

**Comment by pyhalov** [comment]
gmake test fails to compile tests:
```
valgrind-3.12.0/memcheck/tests/badjump2.c:21:21: error: storage size of ‘sigsegv_new’ isn’t know
...
```

**Comment by alarcher** [comment]
Fixed with -D__EXTENSIONS__ but other issues related to asm. I do not know this type of things (yet).

**Comment by alarcher** [comment]
Fixed with -std=gnu99 for asm only defined as GNU extension.

File: components/developer/valgrind/valgrind.p5m [hunk 1]
```diff
@@ -48,18 +48,125 @@ file path=usr/bin/$(MACH64)/valgrind
 file path=usr/bin/$(MACH64)/valgrind-di-server
 file path=usr/bin/$(MACH64)/valgrind-listener
 file path=usr/bin/$(MACH64)/vgdb
-
 file path=usr/include/valgrind/callgrind.h
+file path=usr/include/valgrind/config.h
```
Comment (pyhalov) [inline components/developer/valgrind/valgrind.p5m]: Comment on upper part of valgrind.p5m : what is the point of having isaexec symlinks an delivering only 64-bit version? I'd dropped them and delivered just 64-bit binaries in /usr/bin . 'CONFIGURE_BINDIR.64 = $(CONFIGURE_PREFIX)/bin' can help.

**Review by pyhalov — CHANGES_REQUESTED** [review]

**Comment by alarcher** [comment]
Just dates back to the time when we did not authorize 64bit in /usr/bin. Should be OK now.

File: components/developer/valgrind/Makefile [hunk 2]
```diff
@@ -20,27 +20,24 @@
...
+  sha256:67ca4395b2527247780f36148b084f5743a68ab0c850cb43e4a5b4b012cf76a1
+COMPONENT_ARCHIVE_URL=	http://valgrind.org/downloads/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL =	http://valgrind.org
 COMPONENT_SUMMARY=	Valgrind: instrumentation framework and tools to detect memory and threading problems
 COMPONENT_BUGDB=	developer/debug/valgrind
 COMPONENT_LICENSE=	GPLv2
-COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
-
 
 include $(WS_TOP)/make-rules/prep.mk
```
Comment (pyhalov) [inline components/developer/valgrind/Makefile]: Sorry.. Please, switch to $(WS_MAKE_RULES) :)

**Comment by alarcher** [comment]
Done.

**Review by pyhalov — APPROVED** [review]

**Pull request merged**